### DISCUSSION TITLE
NextJS 13, Server Side Rendering with App Router

### DISCUSSION THREAD

**Discussion by DavidASix** [firebase/firebase-tools#6229] [Q&A]
Hi Firebase team!

**Some context:**
I've built and deployed a basic blogging site Next13 app on Firebase hosting, using Firestore and other services without issue. The site generates static data for all of my pages, and deploys without an issue. My blog post display pages (stored in the Next app as _app/blog/[id]_ are deployed via a cloud function and work well to serve up my SSR content.

I have a page which should list all of the blog posts available, and update as new posts are added to Firestore. As I am using Next13's App routing system, I cannot use _getServerSideProps_, which would [automatically prompt Firebase to create an SSR page](https://firebase.google.com/docs/hosting/frameworks/nextjs#serve_fully_dynamic_content_ssr). 

**The question:**
Am I able to use the new App router data fetching methods and enable SSR via a cloud function? Here is the basic pattern I am looking to implement:

```
async function getBlogPosts() {
  // Get posts from Firestore
}

export default async function BlogPosts() {
  let formulas = await getBlogPosts();
  return (
    // Display blog posts
  );
}
```

Thank you for your time and work you put into this repo and community!

**Environment**

firebase-tools: 12.4.7
next: 13.2.4
firebase: ^9.23.0

**Comment by Ali-Raza764** [comment]
I am also trying to achieve the same using firestore and app router with next js. On researching I found that you can disable caching while using the fetch api as on this link
https://nextjs.org/docs/app/api-reference/functions/fetch
This disables SSG and forces the next js to SSR.
Still I am unable to find how to achieve with firestore. 
There is also a way of revalidation But it is associated with SSG. It re validates data after some time and generates new pages. But no idea how to achieve with firebase. 
I did not answer the problem but I thought that it might help.

**Comment by kirilogan** [comment]
I believe you can accomplish what you are trying to do with:
```
// Opt out of caching for all data requests in the route segment
export const dynamic = 'force-dynamic'
```

See: https://nextjs.org/docs/app/building-your-application/caching#opting-out-1

**Reply by Ali-Raza764** [reply]
Yep I got that after a little researching. I appreciate your Help.
